x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/expression/function/scalar/whitelist/InternalEqlScriptUtils.java

```diff
@@ -33,6 +36,22 @@
     InternalEqlScriptUtils() {
     }
 
+    public static <T> Boolean multiValueDocValues(Map<String, ScriptDocValues<T>> doc, String fieldName, Predicate<T> script) {
+        if (doc.containsKey(fieldName)) {
+            ScriptDocValues<T> docValues = doc.get(fieldName);
+            if (docValues.isEmpty() == false) {
```

Comment:
Should be covered by the for loop and the `docValues.size()` check, no?
Or maybe a check if the `docValues` is `null` could be necessary (if that's a possibility).